### PR TITLE
worked on channel page ui flow

### PR DESCRIPTION
**Pull request by aniekan12**
Users can open a channel, move to the channel info screen and the edit channel info screen

STEPS
1. Click on a channel.
2.  Click on the info iconbutton.
3. click on the edit button on the channel info screen.

https://user-images.githubusercontent.com/45481950/133853610-ba0aa091-2181-4b5e-8e7f-0c1e2094b1d5.mp4

### CODE REVIEW

File: lib/ui/view/channel/channel_view/channel_page_view.dart [hunk 1]
```diff
@@ -176,7 +179,10 @@ AppBar appBar(String text, String nexttext, BuildContext context) {
       Padding(
         padding: const EdgeInsets.only(right: 20.0),
         child: GestureDetector(
-          onTap: () {},
+          onTap: () {
+            Navigator.of(context).push(
```
Comment (isaacadariku) [inline lib/ui/view/channel/channel_view/channel_page_view.dart:183]: Kindly remove this logic from the UI and make use of `model.navigateToChannelView()`

File: lib/ui/view/channel/channel_view/channel_page_viewmodel.dart [hunk 2]
```diff
@@ -1,4 +1,9 @@
+import 'package:hng/app/app.router.dart';
 import 'package:stacked/stacked.dart';
+import 'package:stacked_services/stacked_services.dart';
 
-class ChannelPageViewModel extends BaseViewModel {}
-
+class ChannelPageViewModel extends BaseViewModel {
+  navigateToChannelInfo() {
+    NavigationService().navigateTo(Routes.channelInfoView);
```
Comment (isaacadariku) [inline lib/ui/view/channel/channel_view/channel_page_viewmodel.dart:7]: Here

File: lib/ui/view/channel/channel_info/channel_info_view_model.dart [hunk 3]
```diff
@@ -1,3 +1,10 @@
+import 'package:hng/app/app.locator.dart';
+import 'package:hng/app/app.router.dart';
 import 'package:stacked/stacked.dart';
+import 'package:stacked_services/stacked_services.dart';
 
-class ChannelInfoViewModel extends BaseViewModel {}
+class ChannelInfoViewModel extends BaseViewModel {
+  void navigateToEditChannel() {
+    locator<NavigationService>().navigateTo(Routes.channelInfoEdit);
```
Comment (isaacadariku) [inline lib/ui/view/channel/channel_info/channel_info_view_model.dart:8]: Here

**Review by isaacadariku — CHANGES_REQUESTED** [review]
Good Job!

Can kindly make the `NavigationService()` a variable to be use.

`final _navigationService = locator<NavigationService>();`

File: lib/ui/view/channel/channel_view/channel_page_view.dart [hunk 4]
```diff
@@ -245,7 +249,7 @@ Container channelInfo(String text, String nexttext) {
   );
 }
 
-Row row() {
+Row row(Function()? pressed) {
   final navigator = locator<NavigationService>();
```
Comment (Blazebrain) [inline lib/ui/view/channel/channel_view/channel_page_view.dart:253]: Kindly remove the declaration for the Navigation Service

Registering a service to the locator should come into your viewModel as your view should know nothing as to the logic

**Review by Blazebrain — CHANGES_REQUESTED** [review]
Good to go, once the requested changes have been fixed

File: lib/ui/nav_pages/home_page/home_page_viewmodel.dart [hunk 5]
```diff
@@ -111,6 +111,10 @@ class HomePageViewModel extends StreamViewModel {
     NavigationService().navigateTo(Routes.fileSearchView);
   }
 
+  void navigateToChannelPage() {
+    NavigationService().navigateTo(Routes.channelPageView);
```
Comment (Quwaysim) [inline lib/ui/nav_pages/home_page/home_page_viewmodel.dart:115]: make the NavigationService a field in the class

File: lib/ui/view/forgot_password/forgot_password_otp/forgot_password_otpviewmodel.dart [hunk 6]
```diff
@@ -1,5 +1,4 @@
 import 'dart:async';
-
 import 'package:flutter/material.dart';
```
Comment (Quwaysim) [inline lib/ui/view/forgot_password/forgot_password_otp/forgot_password_otpviewmodel.dart:2]: material import in viewmodel not consistent with stacked, please remove. The stacked way is using [FormViewModel](https://www.filledstacks.com/post/building-flutter-login-and-sign-up-forms/) or check our repo [here](https://github.com/zurichat/zc_app/tree/dev/lib/ui/view/otp) for sample implementation

**Review by Quwaysim — CHANGES_REQUESTED** [review]

File: lib/ui/nav_pages/home_page/home_page_viewmodel.dart (hunk 5 above)
Comment (aniekan12) [reply lib/ui/nav_pages/home_page/home_page_viewmodel.dart]: done

File: lib/ui/view/forgot_password/forgot_password_otp/forgot_password_otpviewmodel.dart (hunk 6 above)
Comment (aniekan12) [reply lib/ui/view/forgot_password/forgot_password_otp/forgot_password_otpviewmodel.dart:2]: i didn't work on that page. i only pulled from the dev branch

Comment (aniekan12) [reply lib/ui/view/forgot_password/forgot_password_otp/forgot_password_otpviewmodel.dart:2]: okay

Comment (aniekan12) [reply lib/ui/view/forgot_password/forgot_password_otp/forgot_password_otpviewmodel.dart]: done

**Review by Blazebrain — DISMISSED** [review]
LGTM

**Review by mariamadebolahamzat — APPROVED** [review]
All good now

**Review by Blazebrain — APPROVED** [review]

**Pull request merged**